modules/bootcamp/detect_landing_pad.py

```
@@ -79,30 +80,40 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
         # ============
 
         # Ultralytics has documentation and examples
-
+        
         # Use the model's predict() method to run inference
         # Parameters of interest:
         # * source
         # * conf
         # * device
         # * verbose
-        predictions = ...
+        predictions = self.__model.predict([image]) 
 
         # Get the Result object
-        prediction = ...
+        prediction = predictions[0]
 
         # Plot the annotated image from the Result object
         # Include the confidence value
-        image_annotated = ...
+        image_annotated = np.copy(image)
 
         # Get the xyxy boxes list from the Boxes object in the Result object
-        boxes_xyxy = ...
+        boxes_xyxy = prediction.boxes.xyxy
 
         # Detach the xyxy boxes to make a copy,
         # move the copy into CPU space,
         # and convert to a numpy array
-        boxes_cpu = ...
-
+        boxes_cpu = boxes_xyxy.cpu().numpy()
+        bounding_boxes = []
+        for box in boxes_cpu:
+            x1, y1, x2, y2 = map(int, box)
+            cv2.rectangle(image_annotated, (x1, y1), (x2, y2), (0, 255, 0), 2)
+            label = f"{x1}, {y1}, {x2}, {y2}"
+            cv2.putText(image_annotated, label, (x1, y1 - 10), cv2.FONT_HERSHEY_SIMPLEX, 0.5, (255, 0, 0), 2)
+            result_box = bounding_box.BoundingBox.create(np.array([x1, y1, x2, y2]))[1]
+            if result_box is not None:
+                bounding_boxes.append(result_box)
```

instead of skipping failed box creations, we should return an empty list of bounding boxes instead. In statistics, when we get invalid data, we delete the entire data point instead of part of a data point. In this case a datapoint is 1 image, and the bounding boxes on that image are the parts of the data point, so we would delete the whole image instead of just the invalid boxes.

---

I’m thinking of returning ([], image_annotated) if the box creation fails. It handles the case of failure by returning an empty list for bounding boxes ([]), but what should I return for image_annotated? If the creation fails, should I just return the original image?